swap_meet/vendor.py

```diff
@@ -1,2 +1,72 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None) -> None:
+        self.inventory = inventory if inventory is not None else []
+
+    def __str__(self, item):
+        f"{self.item}"
+        # return f"my_item: {self.my_item}, their_item: {self.their_item}"
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        self.inventory.remove(item)
+        return item
+
+    def get_by_category(self, category):
+        category_list = [item for item in self.inventory if item.category == category]
+        return category_list
+
+    def swap_items(self, vendor, my_item, their_item):
+        '''If this Vendor's inventory doesn't contain my_item or the friend's inventory doesn't contain their_item,
+    the method returns False'''
+        if my_item not in self.inventory or their_item not in vendor.inventory:
+            return False
+        self.remove(my_item)
+        vendor.add(my_item)
+        vendor.remove(their_item)
+        self.add(their_item)
+        return True
+
+    def swap_first_item(self, friend):
+        if not self.inventory or not friend.inventory:
+            return False
+        self.swap_items(friend, self.inventory[0],friend.inventory[0])
+        return True
```

Comment:
Great use of helper functions! 

Currently, swapping any items has a time complexity of O(n) due to the `remove` method in `swap_items`. We can make a subtle improvement to O(1) by using multiple assignment instead:

```
        self.inventory[0], other.inventory[0] = other.inventory[0], self.inventory[0]
```        

Here's more info on Python multiple assignment: https://www.w3schools.com/python/gloss_python_assign_value_to_multiple_variables.asp